src/write_gcm.c

```diff
@@ -572,23 +572,22 @@ static int wg_curl_get_or_post(char *response_buffer,
 
   long response_code;
   curl_result = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
-  write_ctx.data[0] = 0;
   if (response_code >= 400) {
     if (!silent_failures) {
       WARNING("write_gcm: Unsuccessful HTTP request %ld: %s",
-              response_code, response_buffer);
+              response_code, write_ctx.data);
     }
     result = -2;
     goto leave;
   }
 
-  if (write_ctx.size < 2) {
-    ERROR("write_gcm: wg_curl_get_or_post: The receive buffer overflowed.");
-    DEBUG("write_gcm: wg_curl_get_or_post: Received data is: %s",
-        response_buffer);
+  if (!write_ctx.data) {
```

Comment:
This will only happen if the first realloc fails. See my suggestion below about using `size` to check for this.